topic_table_model.cpp

```diff
@@ -531,15 +531,187 @@ CORBA::TCKind TopicTableModel::typekind_to_tckind(DDS::TypeKind tk)
+        }
+        break;
+    }
+    case CORBA::tk_ushort: {
+        CORBA::UShort value;
+        if (check_rc(data->get_uint16_value(value, id), "get_uint16_value failed")) {
+            data_row->value = static_cast<uint16_t>(value);
+        }
+        break;
+    }
+    case CORBA::tk_ulong: {
+        CORBA::ULong value;
+        if (check_rc(data->get_uint32_value(value, id), "get_uint32_value failed")) {
+            data_row->value = static_cast<uint32_t>(value);
+        }
+        break;
+    }
+    case CORBA::tk_float: {
+        CORBA::Float value;
+        if (check_rc(data->get_float32_value(value, id), "get_float32_value failed")) {
+            data_row->value = static_cast<float>(value);
+        }
+        break;
+    }
+    case CORBA::tk_double: {
+        CORBA::Double value;
+        if (check_rc(data->get_float64_value(value, id), "get_float64_value failed")) {
+            data_row->value = static_cast<double>(value);
+        }
+        break;
+    }
+    case CORBA::tk_boolean: {
+        CORBA::Boolean value;
+        if (check_rc(data->get_boolean_value(value, id), "get_boolean_value failed")) {
+            data_row->value = static_cast<uint32_t>(value);
+        }
+        break;
+    }
+    case CORBA::tk_char: {
+        CORBA::Char value;
+        if (check_rc(data->get_char8_value(value, id), "get_char8_value failed")) {
+            data_row->value = value;
+        }
+        break;
+    }
+    case CORBA::tk_wchar: {
+        CORBA::WChar value;
+        if (check_rc(data->get_char16_value(value, id), "get_char16_value failed")) {
+          // TODO: Set to data_row?
+          // data_row->value = value;
+        }
+        break;
+    }
+    case CORBA::tk_octet: {
+        CORBA::Octet value;
+        if (check_rc(data->get_byte_value(value, id), "get_byte_value failed")) {
+            data_row->value = static_cast<uint8_t>(value);
+        }
+        break;
+    }
+    case CORBA::tk_longlong: {
+        CORBA::LongLong value;
+        if (check_rc(data->get_int64_value(value, id), "get_int64_value failed")) {
+            data_row->value = static_cast<qint64>(value);
+        }
+        break;
+    }
+    case CORBA::tk_ulonglong: {
+        CORBA::ULongLong value;
+        if (check_rc(data->get_uint64_value(value, id), "get_uint64_value failed")) {
+            data_row->value = static_cast<quint64>(value);
+        }
+        break;
+    }
+    case CORBA::tk_string: {
+        CORBA::String_var value;
+        if (check_rc(data->get_string_value(value, id), "get_string_value failed")) {
+            data_row->value = value.in();
+        }
+        break;
+    }
+    case CORBA::tk_enum: {
+        // When @bit_bound is supported, update this to call the right interface.
+        CORBA::Long value;
+        if (check_rc(data->get_int32_value(value, id), "get enum value failed")) {
+            data_row->value = static_cast<int32_t>(value);
```

Comment:
Could this display the name of the enumerator instead of the integer value?  Doesn't need to be in this PR.

Reply:
Yes, the name can be gotten from the `DynamicType` of the enum. I went ahead and add that.